src/api/app/services/rpmlint_log_extractor.rb

```diff
@@ -0,0 +1,56 @@
+class RpmlintLogExtractor
+  attr_reader :project, :package, :repository, :architecture, :log_content
+
+  def initialize(params)
+    @project = params[:project]
+    @package = params[:package]
+    @repository = params[:repository]
+    @architecture = params[:architecture]
+
+    @log_content = ''
+  end
+
+  def call
+    # Retrieve the content of the 'rpmlint.log' file
+    Backend::Api::BuildResults::Binaries.rpmlint_log(project, package, repository, architecture)
```

Comment:
Can we check for the existence of the `rpmlint.log` file instead of using an exception for flow control?

Reply:
Using an exception for flow control is how we check for the existence of the `rpmlint.log` file, now. See the implementation of the `rpmlint_log` method: https://github.com/openSUSE/open-build-service/blob/75a7d0df472b3532a732ec3a7fe79f175b719bd1/src/api/app/lib/backend/api/build_results/binaries.rb#L47-L51

Do you have another approach in mind?